src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -209,14 +215,25 @@ private void OnTimer(object sender, ElapsedEventArgs e)
                 // schedule. Errors will be logged to Dashboard already.
             }
 
+            // If the trigger fired before it was officially scheduled (likely under 1 second due to clock skew),
+            // adjust the invocation time forward for the purposes of calculating the next occurrence.
+            // Without this, it's possible to set the 'Next' value to the same time twice in a row, 
+            // which results in duplicate triggers if the site restarts.
+            DateTime adjustedInvocationTime = invocationTime;
+            if (!isPastDue && !runOnStartup && ScheduleStatus?.Next > invocationTime)
```

Comment:
Why is `!runOnStartup` part of the clause? That would mean that any RunOnStartup functions won't get your fix.